BuildForSDG/SmeVest-Frontend
# feat: implement sign up functionality

- Leverage the sign up api endpoint to enable account creation
- used redux for storage of data like token and emailVerification code
- Designed EmailVerification page
- Created unit tests for auth reducers and actions

Delivers #9

## Description

1. The sign up functionality should be able to leverage the sign up api endpoint to successfully sign up a user
2. The sign up functionality should also be able to give error feedback if there is an error during signing up
3. On successful sign up, the user should be redirected to the email verification page where the user will input the code sent to his/her mail
4. sign up functionality should leverage storing temp data with redux and local storage

## How has this been tested

1. In your terminal, run git clone https://github.com/BuildForSDG/SmeVest-Frontend.git
2. Run cd SmeVest-Frontend
3. Run git fetch origin ft-signup-func
4. Run git checkout origin/ft-signup-func
5. Run yarn install
6. Run yarn start
7. Go to http://localhost:1234/signup on your browser
8. Fill the form and click the signup button
9. On successful signup will take you to http://localhost:1234/verify
10. Run yarn run test to run unit tests

Alternative
none

Additional context
none

## Screenshots (if applicable, else remove this line / section)
![image](https://user-images.githubusercontent.com/27803089/82093301-c3d52800-96f2-11ea-9b91-4c731a3c0de0.png)


## Checklist:
